ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/resources/KsqlResourceTest.java

```diff
@@ -522,6 +524,62 @@ public void shouldReturn5xxOnStatementSystemError() {
     EasyMock.verify(ksqlEngine);
   }
 
+  @Test
+  public void shouldFailIfReachedActivePersistentQueriesLimit() {
```

Comment:
Probably overkill to have both 'reached' and 'exceeded' unit tests in here as you already have this at the unit level of `QueryCapacityUtil`.  Just add one where the new request takes it over the threshold by 1.

However, what might be nice is to have the test run some `ksqlString` which contains multiple query statements. Where the second statement takes it just over the theshold and then ensure:
a) the request failed
b) the first query statement was _not_ started.

Also, consider adding a test that proves you can still run non-persisten-query statements even when the active query limit is at threshold.